backend/src/main/java/com/woowacourse/momo/auth/support/SHA256Encoder.java

```diff
@@ -0,0 +1,34 @@
+package com.woowacourse.momo.auth.support;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.NoSuchElementException;
+
+import org.springframework.stereotype.Component;
+
+import com.woowacourse.momo.member.domain.PasswordEncoder;
+
+@Component
+public class SHA256Encoder implements PasswordEncoder {
+
+    private static final String PASSWORD_ENCRYPT_ALGORITHM = "SHA-256";
+
+    @Override
+    public String encrypt(String text) {
+        try {
+            MessageDigest md = MessageDigest.getInstance(PASSWORD_ENCRYPT_ALGORITHM);
+            md.update(text.getBytes());
+            return bytesToHex(md.digest());
+        } catch (NoSuchAlgorithmException e) {
+            throw new NoSuchElementException("비밀번호 암호화 알고리즘을 찾을 수 없습니다.");
+        }
+    }
```

Comment:
> `MessageDigest.getInstance()`는 `try/catch`로 묶여야해서 상수로 분리가 어려울 것 같아요🥲

이러한 방식을 생각했는데 update 메서드 때문에 인스턴스의 상태가 바뀌니까 안되겠네요..!!

getInstance 메서드를 통해 "SHA-256"에 해당하는 인스턴스를 가져오고 계속 재사용할 줄 알았는데,
인스턴스의 hashcode를 출력하면서 확인해보니까 매번 새로운 인스턴스가 생성되네요!

```suggestion
        private static final MessageDigest MESSAGE_DIGEST;

    static {
        try {
            MESSAGE_DIGEST = MessageDigest.getInstance("SHA-256");
        } catch (NoSuchAlgorithmException e) {
            throw new NoSuchElementException("비밀번호 암호화 알고리즘을 찾을 수 없습니다.");
        }
    }

    @Override
    public String encrypt(String text) {
        MESSAGE_DIGEST.update(text.getBytes());
        return bytesToHex(MESSAGE_DIGEST.digest());
    }
```